### PR TITLE
Media plan storage indicator: add accessible tooltip

### PR DESCRIPTION
**Pull request by frontdevde**
#### Changes proposed in this Pull Request

This PR adds an accessible tooltip to the plan storage indicator in `/media`.

See paYJgx-WR-p2 for use of reakit and a11y considerations.

Fixes https://github.com/Automattic/wp-calypso/issues/40986

somegif

I also noticed that the plan storage block doesn't have a focus style that matches the rest of the buttons in the media library. I changed it to now match the other blocks.

Screenshot 2020-08-25 at 18 56 46

#### Testing instructions

* Using the [calypso.live](https://calypso.live/?branch=update/plan-storage-add-tooltip) link below, navigate to `/media`
* Hover over the storage indicator to show the new tooltip
* Verify it works as expected (see GIF above)

### CODE REVIEW

File: client/assets/stylesheets/shared/functions/_z-index.scss [hunk 1]
```diff
@@ -171,6 +171,7 @@ $z-layers: (
 		'.dialog__backdrop': 100200,
 		'.wplink__dialog.dialog.card': 100200,
 		'.web-preview': 100200,
+		'.plan-storage__tooltip': 100300,
```
Comment (frontdevde) [inline client/assets/stylesheets/shared/functions/_z-index.scss:174]: As the reakit tooltip sits outside the page flow at the end of the page, we need a high enough z-index higher than the media library modal at 100200.

**Comment by obenland** [comment]
I'm not over the moon about introducing a new Tooltip component and not fixing/replacing the existing one, to be honest.

@Automattic/team-calypso Do you have thoughts/preferences on this? Is moving `Tooltip` away from `Popover` even an option at this point?

**Comment by mreishus** [comment]
I noticed that the `./client/` directory of calypso includes [version 1.0.0-beta.14 of reakit](https://github.com/Automattic/wp-calypso/blob/579cc8bb5498f53e3025152fd806e4939df4d10e/client/package.json#L179), when 1.2.3 is the latest. It looks like commit 79cfa4d214fa added the beta.14 version, which never got updated. Then a9fb3b8c83 removed the last use of reakit in `./client/`, but didn't remove the dependency.  This means the changes in this PR are the only use of reakit in `./client`, so an upgrade to the reakit only has to consider the changes done in the PR to make sure they still work.  I think this is a great opportunity to move off 1.0.0-beta.14 if we're still going with reakit.

**Comment by griffbrad** [comment]
I have no issue with the Reakit use here overall. As @mreishus notes, it's already used heavily by `@wordpress/components` and will likely be used elsewhere in Calypso as well. In the context of `Popover` and `Tooltip`, it relies on popper.js (https://popper.js.org/), which provides much better, more sophisticated positioning logic than the current Calypso component. Discussion of using popper.js directly or via Reakit has recently resurfaced in Gutenberg:

https://github.com/WordPress/gutenberg/issues/21275#issuecomment-664511197

I don't want to over-complicate this PR by requiring that we sort all of this out fully before merging. However, the many `!important` declarations in the styles here seem like a code smell and would ultimately just have to be sorted out by anyone attempting to improve `Popover` or `Tooltip` as a whole when they encounter this overly-specific special-case styling and arrow positioning.

@diegohaz Is there a straightforward way to reuse elements of Reakit's positioning and ARIA support in the context of Calypso's existing components? Perhaps via state and/or props hooks?

**Comment by frontdevde** [comment]
> I'm not over the moon about introducing a new Tooltip component and not fixing/replacing the existing one, to be honest.

Agreed. Reading @jeryj's a11y considerations in paYJgx-WR-p2, a more holistic update to all Tooltips in Calypso definitely seems desirable. That said, it would probably be a lot more involved (if only for the amount of testing required to make sure we cover all use cases and don't break anything).

So in the interest of making progress on and solving the specific use case in https://github.com/Automattic/wp-calypso/issues/40986 the idea was to try reakit's Tooltip component as a proof of concept here first. As @mreishus and @griffbrad noted reakit is already used in core and "will likely be used elsewhere in Calypso as well".

> I think this is a great opportunity to move off 1.0.0-beta.14 if we're still going with reakit.

Definitely! Just updated it to the latest stable version.

> However, the many !important declarations in the styles here seem like a code smell and would ultimately just have to be sorted out by anyone attempting to improve Popover or Tooltip as a whole when they encounter this overly-specific special-case styling and arrow positioning.

Agreed, I removed them. FWIW I had used these styles to force the arrow positioning to the right end of the reference element to reflect the design here https://github.com/Automattic/wp-calypso/issues/40986#issuecomment-677707110. 

@diegohaz The placement options `top-start|top-center|top-end` all place the arrow in the top-center of the reference element. As per https://reakit.io/docs/tooltip/#tooltiparrow I would've thought that `top-end` would place the arrow towards the end of the reference element. Am i missing something?

File: client/blocks/plan-storage/tooltip.jsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import React from 'react';
+import {
+	useTooltipState,
+	Tooltip as ReakitTooltip,
+	TooltipReference,
+	TooltipArrow,
+} from 'reakit/Tooltip';
+
+function Tooltip( { children, title, ...props } ) {
+	const tooltip = useTooltipState( { placement: 'top-end', gutter: 14 } );
```
Comment (obenland) [inline client/blocks/plan-storage/tooltip.jsx:13]: I realize Reakit's docs define this constant as `tooltip`, but how would you feel about being a little more descriptive here to differentiate it from all the other Tooltip* mentions? `tooltipState` or something?

**Comment by diegohaz** [comment]
@frontdevde

> The [placement options](https://reakit.io/docs/tooltip/#tooltiparrow) on the tooltip arrow we not working as I expected them to. What's the right way to get the arrow to stick to the bottom of the tooltip (plus when it flips on the top of the tooltip) and point to the end of the reference element as show in this design https://github.com/Automattic/wp-calypso/issues/40986#issuecomment-677707110?

The arrow component expects to receive the `placement` property returned by the `useTooltipState` hook so it can automatically calculate its position and rotate accordingly to the placement of the popover element. So, `placement` always refers to the popover position, not the arrow's.

It's always positioned at the center relatively to the reference element. And as far as I know, Popper.js doesn't give us an option to change this. But I think you can just set a `margin-left` on the arrow element to shift it.

Considering that the arrow width is `1em`, the code below would position the arrow at the right edge of the tooltip. But you can play with this value to get the right position.
```css
.tooltip__arrow {
  margin-left: calc(50% - 0.5em);
}
```

File: client/blocks/plan-storage/tooltip.jsx (hunk 2 above)
Comment (diegohaz) [reply client/blocks/plan-storage/tooltip.jsx:13]: Yep! That's how we name it on Gutenberg too: https://github.com/WordPress/gutenberg/blob/132a01c63e79bfe4514d857f59ed6c39204381f2/packages/components/src/toolbar/toolbar-container.js#L21-L25

**Comment by frontdevde** [comment]
> So, placement always refers to the popover position, not the arrow's. The arrow element is always positioned at the center relatively to the reference element.

I see! In that case the unexpected placements of the arrow when changing the placement to a different value than the reference element's placement was indeed the expected result. :) I would've thought a separate arrow placement would be a feature included in Popper.js.

No worries, moving it with CSS works well in that case then. Thank you, @diegohaz!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/4277552

Thank you @frontdevde for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/4277552).

**Pull request merged**